pkg/dockerutil/client.go

```diff
@@ -72,7 +73,7 @@ func (c *Client) PullImage(ctx context.Context, opts docker.PullImageOptions) er
 		reg = "https://index.docker.io/v1/"
```

Comment:
I think we should move this into the dockerConfigAuthProvider implementation now. Feels like a cleaner place to deal with this.